aiohttp/http_parser.py

```diff
@@ -807,8 +808,13 @@ def feed_eof(self) -> None:
         if self._type == ParseState.PARSE_UNTIL_EOF:
             self.payload.feed_eof()
         elif self._type == ParseState.PARSE_LENGTH:
+            # Calculate how much we received vs expected
+            expected = self._original_length if self._original_length is not None else 0
+            # self._length is decremented as data is received, so received = expected - remaining
+            received = expected - self._length if self._length <= expected else 0
```

Comment:
Is there any way we can get the else condition here? My understanding is that it shouldn't be possible. If we receive the number of bytes expected, then we consider that message complete and read and further bytes as the start of the next message. Therefore, we should never be able to have more than the expected number of bytes (I also don't see any equivalent logic in the Cython changes).